ramalama/model_store.py

```diff
@@ -0,0 +1,184 @@
+import os
+import urllib
+from pathlib import Path
+
+from enum import StrEnum
+from dataclasses import dataclass
+from typing import Dict, Tuple
+
+
+from ramalama.common import verify_checksum, download_file
+
+
+class ModelRegistry(StrEnum):
+    HUGGINGFACE = "huggingface"
+    OLLAMA = "ollama"
+    OCI = "oci"
+    URL = "url"
```

Comment:
Could see the community adding even more here, such as s3:// , just something to keep in mind, I wonder should we continue with https, etc. We do have to code up any protocol we support.

Reply:
Good point! Its already different with the default https download and the huggingface-/ollama-cli - although they use https as well, the "library" is different. Maybe I can inject a class/callable - which specifies HOW to download - to the model stores `new_snapshot` function. Then this would be easily extensible.